logging/impl.go

```diff
@@ -118,7 +118,11 @@ func (imp *impl) AsZap() *zap.SugaredLogger {
 }
 
 func (imp *impl) shouldLog(logLevel Level) bool {
-	return logLevel >= imp.level
+	if GlobalLogLevel.Level() == zapcore.DebugLevel {
```

Comment:
[q] So always log when global log level is DEBUG? Does it not matter if imp's log level is INFO?

Reply:
I probably should have explained that in the PR. The (short-term) goal of the PR was to keep `-debug` doing what it does today. But (medium-term), we'd break from `-debug` behavior and it would be "deprecated" (if you recall that discussion on what I meant by deprecated).

So we're free to remove this line in the future. Not sure how to best broadcast that given there's no concrete criteria.